mat64/dense.go

```diff
@@ -379,6 +379,27 @@ func (m *Dense) TCopy(a Matrix) {
 	*m = w
 }
 
+func (m *Dense) T() {
+	// TODO(jonlawlor): use a fully in-place transpose, without
+	// creating another slice with the same size.
+	// create a new slice
+	d := make([]float64, len(m.mat.Data))
+
+	// copy values into the appropriate location
+	cycle := m.mat.Rows*m.mat.Cols - 1
```

Comment:
I see your point.  I'll have to put together some tests to represent that case, because I'm not used to thinking that way.